### PR TITLE
[REVIEW] Fix result representation in groupby.apply

### PR DESCRIPTION
**Pull request by galipremsagar**
Fixes: #6745, #6788

This PR fixes the result representation of groupby apply function when the custom udf returns a scalar, Series or a DataFrame.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: python/cudf/cudf/core/groupby/groupby.py [hunk 1]
```diff
@@ -415,7 +415,23 @@ def mult(df):
...
             grouped_values[s:e] for s, e in zip(offsets[:-1], offsets[1:])
         ]
-        result = cudf.concat([function(chk) for chk in chunks])
+        chunk_results = [function(chk) for chk in chunks]
+
+        if len(chunk_results) > 0 and cudf.utils.dtypes.is_scalar(
+            chunk_results[0]
+        ):
+            result = cudf.Series(
+                chunk_results, index=self.grouping.keys.unique()
```
Comment (kkraus14) [inline python/cudf/cudf/core/groupby/groupby.py:424]: Does using `self.grouping.keys.unique()` guarantee that the keys are ordered the same as `chunk_results` here? This feels like an implementation detail that we're reliant on.

Comment (galipremsagar) [reply python/cudf/cudf/core/groupby/groupby.py:424]: My bad, yes `unique` seems to call `drop_duplicates` which is not guaranteed to preserve order. Should we build an internal utility something like:
```
   def _ordered_unique(array):
	df = cudf.DataFrame({'a':array, 'b':column.arange(len(array))})
	df = df.drop_duplicates('a')
	df = df.sort_values(by=['b'])
	return df['a']
```

Comment (kkraus14) [reply python/cudf/cudf/core/groupby/groupby.py:424]: I want to say that `self.grouping.keys` should already be sorted in that we should be able to use the offsets to just grab the keys as needed.

Comment (galipremsagar) [reply python/cudf/cudf/core/groupby/groupby.py]: Understood, made the change to extract from offsets

**Review by kkraus14 — APPROVED** [review]

**Comment by kkraus14** [comment]
@shwina can you take a look as well before we merge?

**Review by shwina — APPROVED** [review]
LGTM!

**Pull request merged**